sympy/polys/tests/test_polytools.py

```diff
@@ -3968,3 +3968,20 @@ def test_issue_20985():
     w, R = symbols('w R')
     poly = Poly(1.0 + I*w/R, w, 1/R)
     assert poly.degree() == S(1)
+
+
+def test_issue_23861():
+
+    x = Symbol('x')
+
+    imaginary_str = '3.{}1'.format('0' * 50)
+    imaginary = Float(imaginary_str)
+
+    poly = x + I * imaginary
+    roots = nroots(poly, n=300)
+    assert len(roots) == 1
+
+    root = roots[0]
+    expected = -I * imaginary
+    difference = (root - expected).evalf(300)
+    assert abs(difference) < 1e-300
```

Comment:
Rather than making a new test function this can be added as an example in the existing `test_nroots` function. You can include a link to the issue in a comment.

This test code can also be simplified e.g.:
```python
i = Float('3.000000000000000000000000000000000000000000000000001')
[r] = nroots(x + I*i, n=300)
assert abs(r + I*i) < 1e-300
```